tango/local_workspace.py

```diff
@@ -323,17 +352,21 @@ def find_or_add_step_info(step_or_unique_id: Union[Step, str]) -> StepInfo:
                 # Perform some sanity checks. Sqlite and the file system can get out of sync
                 # when a process dies suddenly.
                 step_dir = self.step_dir(unique_id)
-                new_state = step_info.state
-                if not step_dir.exists() or self.dir_is_empty(step_dir):
-                    new_state = StepState.INCOMPLETE
-                elif step_info.state == StepState.RUNNING and not self._step_lock_file_is_locked(
-                    unique_id
+                step_state_guess = self.guess_step_dir_state(step_dir) or step_info.state
+                if (
+                    step_state_guess == StepState.INCOMPLETE
+                    and step_info.state != StepState.INCOMPLETE
                 ):
-                    new_state = StepState.INCOMPLETE
-                if new_state != step_info.state:
                     step_info.start_time = None
                     step_info.end_time = None
                     d[unique_id] = step_info
```

Comment:
Should we not update `step_info.state` to `INCOMPLETE` is this case?

Reply:
`step_info.state` is a property that becomes `INCOMPLETE` when there is no start time and no end time.